actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -310,8 +304,18 @@ def tag(name = nil, options = nil, open = false, escape = true)
         if name.nil?
           tag_builder
         else
-          name = ERB::Util.xml_name_escape(name) if escape
-          "<#{name}#{tag_builder.tag_options(options, escape) if options}#{open ? ">" : " />"}".html_safe
+          html = tag_builder.public_send(name, escape: escape, **options.to_h)
+          tag_name = name.to_s.dasherize
+          tag_name = ERB::Util.xml_name_escape(tag_name) if escape
+
+          html.delete_suffix!("</#{tag_name}>")
+
+          unless open
+            html.delete_suffix!(">")
+            html.concat(" />")
+          end
+
+          html.html_safe
```

Comment:
This change doesn't make sense to me. Why the framework would do more work than necessary to avoid using its own internal API?

That is why we have `:nodoc`. Methods can have public visibility in Ruby, but they aren't public API.